src/pepr/operator/controllers/istio/egress-ambient.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2025 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+import { V1OwnerReference } from "@kubernetes/client-node";
+import { K8s } from "pepr";
+import { Allow, IstioAuthorizationPolicy, IstioServiceEntry, K8sGateway } from "../../crd";
+import { purgeOrphans } from "../utils";
+import { generateAuthorizationPolicy } from "./auth-policy";
+import { getHostPortsProtocol } from "./egress";
+import { log } from "./istio-resources";
+import { generateLocalEgressSEName, generateLocalEgressServiceEntry } from "./service-entry";
+import { HostResourceMap } from "./types";
+import { createEgressWaypointGateway } from "./ambient-waypoint";
+
+export const ambientEgressNamespace = "istio-egress-waypoint";
+export const sharedEgressPkgId = "shared-ambient-egress-resource";
+
+// Apply the ambient egress resources
+export async function applyAmbientEgressResources(packageList: Set<string>, generation: number) {
+  // If no packages using ambient egress, don't create the waypoint
+  if (packageList.size === 0) {
+    return;
+  }
+
+  // Generate the waypoint payload
+  const waypoint = createEgressWaypointGateway(packageList, generation);
+
+  // Apply waypoint
+  log.debug(waypoint, `Applying Waypoint ${waypoint.metadata?.name}`);
+
+  // Apply the Waypoint and force overwrite any existing resource
+  await K8s(K8sGateway).Apply(waypoint, { force: true });
```

Comment:
This might be okay, but this diverges a bit from some of the logic used for our waypoints with authservice where we also check health after applying ([ref](https://github.com/defenseunicorns/uds-core/blob/main/src/pepr/operator/controllers/istio/ambient-waypoint.ts#L41-L42)). I think there'd be value in a quick call to [`waitForWaypointPodHealthy`](https://github.com/defenseunicorns/uds-core/blob/main/src/pepr/operator/controllers/istio/ambient-waypoint.ts#L152)?